plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -594,6 +636,32 @@ int readRuleStructAndRuleSetFromFile( const char *ruleBaseName, const char *rule
     freeRErrorContent( &errmsgBuf );
     return res;
 }
+int readRuleStructAndRuleSetFromBuffer(const char* ruleBaseName, char* ruleBase)
+{
+    int errloc;
+    rError_t errmsgBuf;
+    errmsgBuf.errMsg = NULL;
+    errmsgBuf.len = 0;
+
+    char* buf = (char*) malloc(ERR_MSG_LEN * 1024 * sizeof(char));
```

Comment:
Let's use a C++ container for memory mgmt or `irods::at_scope_exit` to free the memory. That'll keep this code future proof.